### PR TITLE
Improving edit texts in project creation, fixing localisation bug and add dialogs to open settings and add authorization

### PR DESCRIPTION
**Pull request by alemannosara**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/thegramginers/GPSTracker.java [hunk 1]
```diff
@@ -36,18 +37,27 @@ public Location getLocation() {
...
             isNetworkEnabled = locationManager.isProviderEnabled(locationManager.NETWORK_PROVIDER);
 
-           if (isGPSEnabled) {
-                if (location == null) {
-                    updateLocation();
+            if(ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                    || ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED){
+                if(isGPSEnabled){
+                    if(location == null){
+                        locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 10000, 10, this);
```
Comment (jrossett) [inline app/src/main/java/ch/epfl/sweng/thegramginers/GPSTracker.java]: Why did you change the location of this code?

Comment (alemannosara) [reply app/src/main/java/ch/epfl/sweng/thegramginers/GPSTracker.java]: The updateLocation considered only the NETWORK_PROVIDER and not necessarily the GPS_PROVIDER so that the user location was not appearing on the screen.

File: app/src/main/java/ch/epfl/sweng/thegramginers/OnTagSelectedListenerProject.java [hunk 2]
```diff
@@ -25,14 +25,10 @@ else if(tab.getPosition() == 1){
...
     public void onTabUnselected(TabLayout.Tab tab) {
-        if(tab.getPosition() == 1){
-            switchToNextFragment(new ProjectsListFragment());
-        }
     }
 
     @Override
-    public void onTabReselected(TabLayout.Tab tab) {
-        onTabSelected(tab);
+    public void onTabReselected(TabLayout.Tab tab){
```
Comment (jrossett) [inline app/src/main/java/ch/epfl/sweng/thegramginers/OnTagSelectedListenerProject.java]: Thanks that was breaking my code! (Duplicating the list of projects that was loaded from the database)

Comment (alemannosara) [reply app/src/main/java/ch/epfl/sweng/thegramginers/OnTagSelectedListenerProject.java]: Thanks to you for pointing that out!

File: app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java [hunk 3]
```diff
@@ -19,8 +20,15 @@
...
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
 
 public class ProjectActivityMap extends FragmentActivity implements OnMapReadyCallback {
+    @BindView(R.id.nextMap) FloatingActionButton nextView;
+    @BindView(R.id.localisationOn) FloatingActionButton localisationOn;
+    @BindView(R.id.searchViewMap) EditText searchBar;
+
     protected final static LatLng EPFL = new LatLng(46.51, 6.56);
```
Comment (KnightEther) [inline app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: Why do we have EPFL coordinates?

Comment (alemannosara) [reply app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: I thought it was better to zoom on the EPFL as a default location.

File: app/src/main/res/layout/activity_project_creation_date_salary.xml [hunk 4]
```diff
@@ -10,12 +10,17 @@
     android:orientation="vertical"
     tools:context=".ProjectActivityDateSalary">
 
-    <EditText
-        android:id="@+id/durationProject"
+    <android.support.design.widget.TextInputLayout
```
Comment (KnightEther) [inline app/src/main/res/layout/activity_project_creation_date_salary.xml]: Yeees that's so much better with the input layout <3

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -10,6 +10,15 @@
     <string name="title_profile">Profile</string>
     <string name="profile_name">Profile name</string>
 
+    <string name="project_description">Description</string>
+    <string name="project_name">Project Name</string>
+    <string name="project_new_tag">Enter a new tag here</string>
+    <string name="search">  Search here </string>
+    <string name="project_duration">Duration</string>
```
Comment (jrossett) [inline app/src/main/res/values/strings.xml]: What does this change?

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -10,6 +10,15 @@
...
+    <string name="project_salary">Salary (per hour)</string>
+    <string name="all_the_jobs">All the jobs</string>
+    <string name="on_the_map">On the map</string>
```
Comment (KnightEther) [inline app/src/main/res/values/strings.xml]: Good we will be able to change easier of langage if we need, do you think to add the warning message as well?

File: app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java (hunk 3 above)
Comment (KnightEther) [reply app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: Why not zooming on the user position?

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (KnightEther) [reply app/src/main/res/values/strings.xml]: Because we can easily change of language if we want to, we could just have to traduce this file

File: app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java (hunk 3 above)
Comment (alemannosara) [reply app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: Well..Actually the user may not have allow the GPS so as default, it could be better to have EPFL and then, when clicking on the localisation button, we zoom on his position.

File: app/src/main/res/values/strings.xml (hunk 6 above)
Comment (alemannosara) [reply app/src/main/res/values/strings.xml]: Thanks for pointing this out! I will inspect my code before merging it to master.

File: app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java (hunk 3 above)
Comment (KnightEther) [reply app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: Maybe we can zoom on user if we have the autorisation and if not zoom on EPFL

Comment (alemannosara) [reply app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: Good idea. I juste commit a new version that does that.

Comment (KnightEther) [reply app/src/main/java/ch/epfl/sweng/thegramginers/ProjectActivityMap.java]: Awesome!
